### PR TITLE
QE: BV Fix SLE Micro and improve activation key step

### PR DESCRIPTION
**Pull request by nodeg**
Once reviewed the commits will get sorted again so that each change has its own commit. Furthermore some of the commits will get backported to 4.3/4.2.

## What does this PR change?

This 

- includes more fixes for SLE Micro
- fixes the channel assignment of SLE Micro. During testing SUMA 4.3.5 in BV, I found out that the channel assignment was wrong
wrong channel assignments:  
image
getting the correct labels:
```bash
suma-bv-43-srv:~ # mgr-create-bootstrap-repo 
(...)
10. SLE-MICRO-5.2-x86_64
11. SLE-MICRO-5.3-x86_64
(...)
```
  image
  image
  image
- improves the activation key creation via API by adding some debugging output. This helped me a lot in debugging issues in BV
- add some remarks to some of the constants in `constants.rb`. This will help us in the future when adding a new product to e.g. BV

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests were edited

- [x] **DONE**

## Links

- Manager 4.3: 
- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by nodeg** [comment]
Once reviewed the commits will get sorted again so that each change has its own commit.

File: testsuite/features/step_definitions/api_common.rb [hunk 1]
```diff
@@ -234,15 +234,19 @@
...
   base_channel = LABEL_BY_BASE_CHANNEL[BASE_CHANNEL_BY_CLIENT[client]]
   key = $api_test.activationkey.create(id, description, base_channel, 100)
-  raise if key.nil?
+  raise 'Error creating activation key via API' if key.nil?
+  STDOUT.puts "key #{key} created" unless key.nil?
 
   is_ssh_minion = client.include? 'ssh_minion'
-  $api_test.activationkey.set_details(key, description, base_channel, 100, is_ssh_minion ? 'ssh-push' : 'default')
+  result_details = $api_test.activationkey.set_details(key, description, base_channel, 100, is_ssh_minion ? 'ssh-push' : 'default')
+  raise "Cannot update the details of activation key #{key}" unless result_details == 1
```
Comment (nodeg) [inline testsuite/features/step_definitions/api_common.rb:242]: See https://documentation.suse.com/suma/4.3/api/suse-manager/api/activationkey.html#apidoc-activationkey-setDetails-1880527972

File: testsuite/features/step_definitions/api_common.rb [hunk 2]
```diff
@@ -270,7 +274,8 @@
   child_channels.push(custom_channel)
 
   # Add child channels to the key
-  $api_test.activationkey.add_child_channels(key, child_channels)
+  result_channels = $api_test.activationkey.add_child_channels(key, child_channels)
+  raise "Cannot add child channels to activation key #{key}" unless result_channels == 1
```
Comment (nodeg) [inline testsuite/features/step_definitions/api_common.rb:278]: See https://documentation.suse.com/suma/4.3/api/suse-manager/api/activationkey.html#apidoc-activationkey-addChildChannels-1868933720

File: testsuite/features/support/constants.rb [hunk 3]
```diff
@@ -86,10 +88,10 @@
...
                       'sle15sp4_minion' => 'bison',
                       'sle15sp4_ssh_minion' => 'bison',
-                      'slemicro52_minion' => 'bison',
-                      'slemicro52_ssh_minion' => 'bison',
-                      'slemicro53_minion' => 'bison',
-                      'slemicro53_ssh_minion' => 'bison',
+                      'slemicro52_minion' => 'ethtool',
+                      'slemicro52_ssh_minion' => 'ethtool',
+                      'slemicro53_minion' => 'ethtool',
+                      'slemicro53_ssh_minion' => 'ethtool',
```
Comment (nodeg) [inline testsuite/features/support/constants.rb:94]: We need a different packages fro SLE Micro since it does not have either `autoconf` nor `bison`.

**Review by Bischoff — APPROVED** [review]
thanks a lot!

File: testsuite/features/step_definitions/api_common.rb [hunk 4]
```diff
@@ -234,7 +234,8 @@
   id = description = "#{client}_key"
   base_channel = LABEL_BY_BASE_CHANNEL[BASE_CHANNEL_BY_CLIENT[client]]
   key = $api_test.activationkey.create(id, description, base_channel, 100)
-  raise if key.nil?
+  raise 'Error creating activation key via the API' if key.nil?
```
Comment (vandabarata) [inline testsuite/features/step_definitions/api_common.rb:237]: I thought we were trying to do 
```suggestion
  raise StandardError, 'Error creating activation key via the API' if key.nil?
```
instead?

**Comment by nodeg** [comment]
I rearranged the commits again to be able to port the changes easier. Merging now.

**Pull request merged**